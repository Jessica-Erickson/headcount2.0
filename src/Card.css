.Card {
  align-content: center;
  display: flex;
  flex-direction: column;
  height: 250px;
  justify-content: space-evenly;
  margin: 1rem;
  width: 200px;
}

ul {
  display: inline-block;
  list-style: none;
  margin: .5rem auto;
  padding: 0;
}

h2, h3 {
  margin: 0;
  text-align: center;
}