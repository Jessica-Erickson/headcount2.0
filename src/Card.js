import React from 'react';
import './Card.css';
import PropTypes from 'prop-types';

export default function Card({ data , handleCompareClick }) {
  let card;
  if (handleCompareClick) {
    card = (
      <article onClick={() => {handleCompareClick(data.location)}}>
        <h3>{data.location}</h3>
        {
          Object.keys(data.stats).map(stat => {
            return (
              <p>{stat + ': ' + data.stats[stat]}</p>
            )
          })
        }
      </article>
    );
  } else {
    const names = Object.keys(data)
    card = (
      <article>
        <h2>{names[0] + ': ' + data[names[0]]}</h2>
        <h2>{'<---' + data.compared + '--->'}</h2>
        <h2>{names[1] + ': ' + data[names[1]]}</h2>
      </article>
    )
  }
  return card;
}

Card.propTypes = {
  data: PropTypes.object.isRequired,
  handleCompareClick: PropTypes.func
}