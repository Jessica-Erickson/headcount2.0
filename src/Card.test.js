import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import {shallow} from 'enzyme';

describe('Card', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card data={{}} />, div);
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<Card data={{}} />);

    expect(wrapper).toMatchSnapshot()
  });
});
