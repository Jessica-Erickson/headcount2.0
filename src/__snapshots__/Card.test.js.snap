// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<article>
  <h2>
    undefined: undefined
  </h2>
  <h2>
    &lt;---undefined---&gt;
  </h2>
  <h2>
    undefined: undefined
  </h2>
</article>
`;
